crates/rome_js_analyze/src/analyzers/nursery/no_prototype_builtins.rs

```diff
@@ -0,0 +1,150 @@
+                        let token = literal_expr.value_token().ok()?;
+
+                        is_prototype_builtins(token_text.text()).then_some(RuleState {
+                            prototype_builtins_method_name: token_text.to_string(),
+                            text_range: token.text_range(),
+                        })
+                    }
+                    AnyJsExpression::JsTemplateExpression(expr) => {
+                        let template_element = expr.as_fields().elements.first()?;
+                        let template_chunk_token = template_element
+                            .as_js_template_chunk_element()?
+                            .as_fields()
+                            .template_chunk_token
+                            .ok()?;
+                        let token_text = template_chunk_token.text();
+                        is_prototype_builtins(token_text).then_some(RuleState {
+                            prototype_builtins_method_name: token_text.to_string(),
+                            text_range: template_chunk_token.text_trimmed_range(),
+                        })
+                    }
+                    _ => None,
+                }
+            }
+            AnyJsExpression::JsStaticMemberExpression(expr) => {
+                let member = expr.as_fields().member.ok()?;
+                let value_token = member.as_js_name()?.value_token().ok()?;
+                let token_text = value_token.text();
+                is_prototype_builtins(token_text).then_some(RuleState {
+                    prototype_builtins_method_name: token_text.to_string(),
+                    text_range: value_token.text_range(),
+                })
+            }
+            _ => None,
+        }
+    }
+
+    fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let diag = RuleDiagnostic::new(
+            rule_category!(),
+            state.text_range,
+            markup! {
+                "Do not access Object.prototype method '"{&state.prototype_builtins_method_name}"' from target object."
+            },
+        );
+
+        if state.prototype_builtins_method_name == "hasOwnProperty" {
+            Some(
+                diag.note(markup! {
+                    "Recommended using "<Emphasis>"Object.hasOwn()"</Emphasis>" over direct use of `Object.prototype` builtins and using "<Emphasis>"Object.hasOwnProperty()"</Emphasis>"."
+                })
+                .note(markup! {
+                    "See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn"
+                }),
+            )
+        } else {
+            Some(diag)
+        }
+    }
+}
+
+/// Chekcks if the `Object.prototype` builtins called directly.
+fn is_prototype_builtins(token_text: &str) -> bool {
+    let disallowed_methods = ["hasOwnProperty", "isPrototypeOf", "propertyIsEnumerable"];
```

Comment:
You should move this outside of the function. They are recreated every time you call the function, which is not ideal. 

Or just use a `matches!` macro:
```suggestion
return matches!(token_text, "hasOwnProperty" | "isPrototypeOf" | "propertyIsEnumerable")
```